### DISCUSSION TITLE
Federation addresses for questers

### DISCUSSION THREAD

**Discussion by hanseartic** [kalepail/stellar-quest-bounties#78] [Bounty Ideas]
# Scenario
Currently there are already a few different use-cases where a user's stellar-account is required. The most obvious ones: quests and bounties. But there is more like tip-bot, quest-monitor-bot.

So each time a quest-related service system is established there is need to map users (either discord or github) to stellar accounts. The federation-address would look like `<username>*quest.stellar.org` then.

It would be nice to provide questers with federation addresses for two reasons
* central place where addresses could be looked up
* foster community feeling (is this paying into this?)

# Bounty idea
The goal is to create come up with a bounty that addresses the scenario described above.

For example it could be to create a service that provides login via github or discord and allows users to add their stellar account. Said service could then act as source for a federation server (e.g. stellar's [reference implementation]) linked in https://quest.stellar.org/.well-known/stellar.toml.

## Useful links
- [federation] on stellar's developer docs
- lumenauts explaining [federated addresses]

[federation]: https://developers.stellar.org/docs/glossary/federation/
[federated addresses]: https://www.lumenauts.com/guides/what-are-federated-stellar-addresses
[reference implementation]: https://github.com/stellar/go/tree/master/services/federation

**Comment by hanseartic** [comment]
Before going deeper into the details of a potential bounty let's maybe discuss *if* this scenario actually adds value to the (quest-)ecosystem.

**Comment by rahimklaber** [comment]
I think this is a good idea, Aside from making it easier for Stellar Quest to get deal with addressess, I don't see more uses for it. But maybe we can find something in the future.  And even if we don't end up using it for Stellar Quest it is still a good bounty

**Comment by kalepail** [comment]
A federation server bounty would be great, let's focus on that, and then if we want to expand to using it for Stellar Quest great. Either way it's a great bounty. Maybe encourage doing it via Cloudflare Workers, their KV store and ease of setup would make building a federation server super simple.

Great bounty suggestion!

**Reply by hanseartic** [reply]
Great direction, thx. I will write sth. up.

**Reply by hanseartic** [reply]
#110
